internal/globalconfig/globalconfig.go

```diff
@@ -62,3 +64,31 @@ func RuntimeID() string {
 	defer cfg.mu.RUnlock()
 	return cfg.runtimeID
 }
+
+// HeaderTag returns the tag assigned to the given header
+func HeaderTag(header string) (tag string, ok bool) {
+	cfg.mu.RLock()
+	defer cfg.mu.RUnlock()
+	tag, ok = cfg.headersAsTags[header]
+	return tag, ok
+}
+
+// SetHeaderTag adds config for header `from` with tag value `to`
+func SetHeaderTag(from, to string) {
+	cfg.mu.Lock()
+	defer cfg.mu.Unlock()
+	cfg.headersAsTags[from] = to
+}
+
+// HeaderTagsLen returns the length of globalconfig's headersAsTags map, 0 for empty map
+func HeaderTagsLen() int {
+	cfg.mu.RLock()
+	defer cfg.mu.RUnlock()
+	return len(cfg.headersAsTags)
+}
+
+// ClearHeaderTags assigns headersAsTags to a new, empty map
+// It is invoked when WithHeaderTags is called, in order to overwrite the config
+func ClearHeaderTags() {
+	cfg.headersAsTags = make(map[string]string)
+}
```

Comment:
@katiehockman , in the case that a user configured `DD_TRACE_HEADER_TAGS` and `WithHeaderTags()`, the contrib option should take precedence. So it needs to wipe out the config that from the env var, hence the need for this function.